azurerm/helpers/validate/strings_test.go

```diff
@@ -0,0 +1,67 @@
+package validate
+
+import "testing"
+
+func TestNoEmptyStrings(t *testing.T) {
+	cases := []struct {
+		Value    string
+		TestName string
+		ErrCount int
+	}{
+		{
+			Value:    "!",
+			TestName: "Exclamation",
+			ErrCount: 0,
+		},
+		{
+			Value:    ".",
+			TestName: "Period",
+			ErrCount: 0,
+		},
+		{
+			Value:    "-",
+			TestName: "Hyphen",
+			ErrCount: 0,
+		},
+		{
+			Value:    "_",
+			TestName: "Underscore",
+			ErrCount: 0,
+		},
+		{
+			Value:    "10.1.0.0/16",
+			TestName: "IP",
+			ErrCount: 0,
+		},
+		{
+			Value:    "",
+			TestName: "Empty",
+			ErrCount: 1,
+		},
+		{
+			Value:    " ",
+			TestName: "Space",
+			ErrCount: 1,
+		},
+		{
+			Value:    "  1",
+			TestName: "DoubleSpaceOne",
+			ErrCount: 0,
+		},
+		{
+			Value:    "1 ",
+			TestName: "OneSpace",
+			ErrCount: 0,
+		},
```

Comment:
it'd be good to add a test case for 5 empty spaces in here too

Reply:
Doen.